### ISSUE TITLE
NZXT RGB & Fan Controller Timeout

### ISSUE THREAD

**Issue by codifryed** [liquidctl/liquidctl#526]
Hi :wave: 

When the hwmon kernel driver is not present or used, the NZXT RGB & Fan Controller needs longer than 2 seconds to complete a read during initialization, causing a Timeout error with liquidctl 1.11.0. This is repeatable and confirmed by another user as well.

```
liquidctl --version              
liquidctl v1.11.0 (Linux-6.0.2-zen1-1-zen-x86_64-with-glibc2.36)

liquidctl -m "rgb" initialize
NZXT RGB & Fan Controller
└── Firmware version    1.13.0

liquidctl -m "rgb" initialize --direct-access
WARNING: forcing re-initialization despite nzxt-smart2 kernel driver
ERROR: unexpected error with NZXT RGB & Fan Controller: Operation timed out
```

When running the same command with 1.10.0:
```
liquidctl --version
liquidctl v1.10.0 (Linux-6.0.2-zen1-1-zen-x86_64-with-glibc2.36)

liquidctl -m "rgb" initialize --direct-access
WARNING: forcing re-initialization despite nzxt_smart2 kernel driver
NZXT RGB & Fan Controller
└── Firmware version    1.13.0 
```

Using Coolero I was able to determine the approximate time needed for initialization with Liquidctl 1.10.0 
```
2022-10-18 21:05:16,260 DEBUG: liquidctl.driver.usb - discarded 0 previously enqueued reports
2022-10-18 21:05:16,260 WARNING: liquidctl.driver.smart_device - forcing re-initialization despite nzxt_smart2 kernel driver
2022-10-18 21:05:16,260 DEBUG: liquidctl.driver.usb - writting report 0x60 with 63 bytes: 02:01:e8:01:01:e8:01:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
2022-10-18 21:05:16,261 DEBUG: liquidctl.driver.usb - writting report 0x60 with 63 bytes: 03:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
2022-10-18 21:05:16,262 DEBUG: liquidctl.driver.usb - writting report 0x10 with 63 bytes: 01:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
2022-10-18 21:05:16,263 DEBUG: liquidctl.driver.usb - writting report 0x20 with 63 bytes: 03:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
2022-10-18 21:05:16,264 DEBUG: liquidctl.driver.usb - read 64 bytes: 11:01:37:00:1d:00:19:51:31:31:32:33:31:37:10:20:01:01:0d:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
2022-10-18 21:05:18,670 DEBUG: liquidctl.driver.usb - read 64 bytes: 21:03:37:00:1d:00:19:51:31:31:32:33:31:37:02:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
2022-10-18 21:05:18,670 DEBUG: coolero.repositories.liquidctl_repo - Liquidctl device initialization response: [('Firmware version', '1.13.0', '')]
``` 

I repeated this several times with the following read times:
(seconds)
2.457
2.459
2.316
2.293
2.406

This came to my attention from an issue submitted by a Coolero user, who doesn't have the kernel driver installed: https://gitlab.com/coolero/coolero/-/issues/74

Liquidctl Debug output:

1.11.0
```
liquidctl -m "rgb" initialize --direct-access --debug
[DEBUG] (cli) (main): script: /usr/bin/liquidctl
[DEBUG] (cli) (main): version: 1.11.0
[DEBUG] (cli) (main): platform: Linux-6.0.2-zen1-1-zen-x86_64-with-glibc2.36
[DEBUG] (cli) (_log_requirements): python: 3.10.8 (main, Oct 11 2022, 20:04:56) [GCC 12.2.0]
[DEBUG] (cli) (_log_requirements): colorlog: 6.7.0
[DEBUG] (cli) (_log_requirements): crcmod: 1.7
[DEBUG] (cli) (_log_requirements): docopt: 0.6.2
[DEBUG] (cli) (_log_requirements): hidapi: 0.12.0.post2
[DEBUG] (cli) (_log_requirements): pyusb: 1.2.1
[DEBUG] (cli) (_log_requirements): pillow: 9.2.0
[DEBUG] (cli) (_log_requirements): smbus: 1.1
[DEBUG] (cli) (_log_requirements): winusbcdc: version n/a (No package metadata was found for winusbcdc)
[DEBUG] (cli) (_log_requirements): libusb-package: version n/a (No package metadata was found for libusb-package)
[DEBUG] (smbus) (find_devices): searching LinuxI2c
[DEBUG] (smbus) (find_devices): LinuxI2c drivers: Ddr4Temperature, EvgaPascal, RogTuring, VengeanceRgb
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-3
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-1
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-8
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-6
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-4
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-2
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-0
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-7
[DEBUG] (smbus) (find_devices): I²C adapter: 6-0008 (skipped, not a bus or unsupported adapter)
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-5
[DEBUG] (usb) (find_devices): searching HidapiBus
[DEBUG] (usb) (find_devices): HidapiBus drivers: Aquacomputer, AuraLed, CommanderCore, CommanderPro, CorsairHidPsu, H1V2, HydroPlatinum, Kraken2, KrakenX3, KrakenZ3, NzxtEPsu, RgbFusion2, SmartDevice, SmartDevice2, _BaseSmartDevice
[DEBUG] (usb) (find_devices): HID device: 1852:7022 (usage_page=0x000c usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 3434:0280 (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 3434:0280 (usage_page=0xff60 usage=0x0061)
[DEBUG] (usb) (find_devices): HID device: 3434:0280 (usage_page=0x0001 usage=0x0080)
[DEBUG] (usb) (find_devices): HID device: 3434:0280 (usage_page=0x000c usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 3434:0280 (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0x0001 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0x0001 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0x000c usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0xff00 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0xff00 usage=0x0004)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0x0001 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0x0001 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0xff00 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0xff00 usage=0x0004)
[DEBUG] (usb) (find_devices): HID device: 046d:c52b (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:c52b (usage_page=0xff00 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:c52b (usage_page=0xff00 usage=0x0004)
[DEBUG] (usb) (find_devices): HID device: 1462:7c37 (usage_page=0x0001 usage=0x0000)
[DEBUG] (usb) (find_devices): HID device: 1462:7c37 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 1e71:2010 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (__init__): has kernel driver: nzxt-smart2 (/sys/class/hidraw/hidraw3/device/hwmon/hwmon5)
[DEBUG] (usb) (probe): SmartDevice2 identified: NZXT RGB & Fan Controller
[DEBUG] (usb) (find_devices): HID device: 1e71:2007 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 2708:0009 (usage_page=0x0001 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 2708:0009 (usage_page=0x0001 usage=0x0001)
[DEBUG] (usb) (find_devices): searching PyUsbBus
[DEBUG] (usb) (find_devices): PyUsbBus drivers: Hydro690Lc, HydroPro, Legacy690Lc, Modern690Lc, _Base690Lc, _ModernBase690Lc
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 2708:0009
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 1e71:2007
[DEBUG] (usb) (find_devices): USB device: 1e71:2010
[DEBUG] (usb) (find_devices): USB device: 05e3:0608
[DEBUG] (usb) (find_devices): USB device: 1462:7c37
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 046d:c52b
[DEBUG] (usb) (find_devices): USB device: 3434:0280
[DEBUG] (usb) (find_devices): USB device: 046d:0843
[DEBUG] (usb) (find_devices): USB device: 1852:7022
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
[DEBUG] (cli) (main): device: NZXT RGB & Fan Controller
[DEBUG] (usb) (clear_enqueued_reports): discarded 0 previously enqueued reports
[WARNING] (smart_device) (initialize): forcing re-initialization despite nzxt-smart2 kernel driver
[DEBUG] (usb) (write): writting report 0x60 with 63 bytes: 02:01:e8:01:01:e8:01:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (write): writting report 0x60 with 63 bytes: 03:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (write): writting report 0x10 with 63 bytes: 01:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (write): writting report 0x20 with 63 bytes: 03:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (read): read 64 bytes: 67:04:37:00:1d:00:19:51:31:31:32:33:31:37:03:00:00:00:00:00:00:00:00:00:c0:2f:fc:2f:1a:30:00:00:00:00:00:00:00:00:00:00:04:00:01:00:01:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (read): read 64 bytes: 11:01:37:00:1d:00:19:51:31:31:32:33:31:37:10:20:01:01:0d:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (read): failed to read, timed out after 2000 ms
[INFO] (cli) (log): detailed error: unexpected error with NZXT RGB & Fan Controller: Operation timed out
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/liquidctl/cli.py", line 460, in main
    status = dev.initialize(**opts)
  File "/usr/lib/python3.10/site-packages/liquidctl/driver/smart_device.py", line 549, in initialize
    self._read_until(parsers)
  File "/usr/lib/python3.10/site-packages/liquidctl/driver/smart_device.py", line 608, in _read_until
    msg = self.device.read(self._READ_LENGTH)
  File "/usr/lib/python3.10/site-packages/liquidctl/driver/usb.py", line 461, in read
    raise Timeout()
liquidctl.error.Timeout
[ERROR] (cli) (log): unexpected error with NZXT RGB & Fan Controller: Operation timed out
```

1.10.0
```
liquidctl -m "rgb" initialize --direct-access --debug
[DEBUG] (cli) (main): liquidctl: 1.10.0
[DEBUG] (cli) (main): platform: Linux-6.0.2-zen1-1-zen-x86_64-with-glibc2.36
[DEBUG] (cli) (_log_requirements): python: 3.10.8 (main, Oct 11 2022, 20:04:56) [GCC 12.2.0]
[DEBUG] (cli) (_log_requirements): colorlog: 6.7.0
[DEBUG] (cli) (_log_requirements): crcmod: 1.7
[DEBUG] (cli) (_log_requirements): docopt: 0.6.2
[DEBUG] (cli) (_log_requirements): hidapi: 0.12.0.post2
[DEBUG] (cli) (_log_requirements): pyusb: 1.2.1
[DEBUG] (cli) (_log_requirements): pillow: 9.2.0
[DEBUG] (cli) (_log_requirements): smbus: 1.1
[DEBUG] (cli) (_log_requirements): winusbcdc: version n/a (No package metadata was found for winusbcdc)
[DEBUG] (cli) (_log_requirements): libusb-package: version n/a (No package metadata was found for libusb-package)
[DEBUG] (smbus) (find_devices): searching LinuxI2c
[DEBUG] (smbus) (find_devices): LinuxI2c drivers: Ddr4Temperature, EvgaPascal, RogTuring, VengeanceRgb
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-3
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-1
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-8
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-6
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-4
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-2
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-0
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-7
[DEBUG] (smbus) (find_devices): I²C adapter: 6-0008 (skipped, not a bus or unsupported adapter)
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-5
[DEBUG] (usb) (find_devices): searching HidapiBus
[DEBUG] (usb) (find_devices): HidapiBus drivers: AuraLed, CommanderCore, CommanderPro, CorsairHidPsu, H1V2, HydroPlatinum, Kraken2, KrakenX3, KrakenZ3, NzxtEPsu, RgbFusion2, SmartDevice, SmartDevice2, _BaseSmartDevice
[DEBUG] (usb) (find_devices): HID device: 1852:7022 (usage_page=0x000c usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 3434:0280 (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 3434:0280 (usage_page=0xff60 usage=0x0061)
[DEBUG] (usb) (find_devices): HID device: 3434:0280 (usage_page=0x0001 usage=0x0080)
[DEBUG] (usb) (find_devices): HID device: 3434:0280 (usage_page=0x000c usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 3434:0280 (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0x0001 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0x0001 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0x000c usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0xff00 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:407b (usage_page=0xff00 usage=0x0004)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0x0001 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0x0001 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0xff00 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:4082 (usage_page=0xff00 usage=0x0004)
[DEBUG] (usb) (find_devices): HID device: 046d:c52b (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:c52b (usage_page=0xff00 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:c52b (usage_page=0xff00 usage=0x0004)
[DEBUG] (usb) (find_devices): HID device: 1462:7c37 (usage_page=0x0001 usage=0x0000)
[DEBUG] (usb) (find_devices): HID device: 1462:7c37 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 1e71:2010 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (__init__): has kernel driver: nzxt_smart2 (/sys/class/hidraw/hidraw3/device/hwmon/hwmon5)
[DEBUG] (usb) (probe): SmartDevice2 identified: NZXT RGB & Fan Controller
[DEBUG] (usb) (find_devices): HID device: 1e71:2007 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 2708:0009 (usage_page=0x0001 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 2708:0009 (usage_page=0x0001 usage=0x0001)
[DEBUG] (usb) (find_devices): searching PyUsbBus
[DEBUG] (usb) (find_devices): PyUsbBus drivers: Hydro690Lc, HydroPro, Legacy690Lc, Modern690Lc, _Base690Lc, _ModernBase690Lc
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 2708:0009
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 1e71:2007
[DEBUG] (usb) (find_devices): USB device: 1e71:2010
[DEBUG] (usb) (find_devices): USB device: 05e3:0608
[DEBUG] (usb) (find_devices): USB device: 1462:7c37
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 046d:c52b
[DEBUG] (usb) (find_devices): USB device: 3434:0280
[DEBUG] (usb) (find_devices): USB device: 046d:0843
[DEBUG] (usb) (find_devices): USB device: 1852:7022
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
[DEBUG] (cli) (main): device: NZXT RGB & Fan Controller
[DEBUG] (usb) (clear_enqueued_reports): discarded 0 previously enqueued reports
[WARNING] (smart_device) (initialize): forcing re-initialization despite nzxt_smart2 kernel driver
[DEBUG] (usb) (write): writting report 0x60 with 63 bytes: 02:01:e8:01:01:e8:01:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (write): writting report 0x60 with 63 bytes: 03:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (write): writting report 0x10 with 63 bytes: 01:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (write): writting report 0x20 with 63 bytes: 03:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (read): read 64 bytes: 11:01:37:00:1d:00:19:51:31:31:32:33:31:37:10:20:01:01:0d:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (read): read 64 bytes: 61:03:37:00:1d:00:19:51:31:31:32:33:31:37:03:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
[DEBUG] (usb) (read): read 64 bytes: 21:03:37:00:1d:00:19:51:31:31:32:33:31:37:02:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
NZXT RGB & Fan Controller
└── Firmware version    1.13.0 
```

---

Additional information about the system:

- Affected device(s): NZXT RGB & Fan Controller
- Operating system and version: Arch Linux
- Liquidctl installation method: Pip, Arch Linux
- Output of `liquidctl --version`: liquidctl v1.11.0 (Linux-6.0.2-zen1-1-zen-x86_64-with-glibc2.36)

**Comment by jonasmalacofilho** [comment]
Hey!  First, sorry about the breakage...

Next, when I first read this report I was worried that there might be something wrong the `update_interval` on those devices.  But that doesn't seem to be the case, instead, it seems that only this specific response (lighting infos) takes a bit longer, and that makes sense.

While we could just have this specific driver use a longer timeout, I think it's also reasonable to increase the default timeout, as 2 seconds was indeed maybe a bit too optimistic: for reading periodic responses 2 seconds is probably fine, but certain initialization commands (and other writes) can take a bit longer.

Can you try 066488c767c3e5489235fa52b7d06b5f4b6d87f7 and confirm that it fixes the issue?

**Comment by codifryed** [comment]
Thanks @jonasmalacofilho,
All good, I know it's pretty hard to know max response times for all these devices. Sometimes we start out optimistic. :wink: 

Fix tested successfully: (total command completion time was always < 3 seconds)
```
liquidctl --version
liquidctl v1.12.0.dev15+g066488c (Linux-6.0.2-zen1-1-zen-x86_64-with-glibc2.36)

liquidctl -m "rgb" initialize --direct-access        
WARNING: forcing re-initialization despite nzxt-smart2 kernel driver
NZXT RGB & Fan Controller
└── Firmware version    1.13.0
```

**Comment by jonasmalacofilho** [comment]
@codifryed ,

Thanks!  Patch release will almost certainly go out later today.

> (total command completion time was always < 3 seconds)

Yeah, I really couldn't see any reason why the one digit fix would not work.  But maybe something else was also broken; better to wait for the confirmation than to make another broken release.

**Comment by jonasmalacofilho** [comment]
Fixed in [liquidctl 1.11.1](https://github.com/liquidctl/liquidctl/releases/tag/v1.11.1).
